File: src/main/java/com/haozileung/infra/web/BaseFilter.java
package com.haozileung.infra.web;

import javax.servlet.Filter;
import javax.servlet.FilterConfig;
import javax.servlet.ServletException;

public abstract class BaseFilter implements Filter {

    @Override
    public void init(FilterConfig filterConfig) throws ServletException {
        if (Initializer.getInjector() != null) {
            Initializer.getInjector().injectMembers(this);
        }
    }
}

File: src/main/kotlin/com/haozileung/infra/web/BaseFilter.kt
package com.haozileung.infra.web;

import com.google.inject.Injector
import javax.servlet.Filter
import javax.servlet.FilterConfig

abstract class BaseFilter : Filter {

    override fun init(filterConfig: FilterConfig) {
        if (Initializer.instance != null) {
            (Initializer.instance as Injector).injectMembers(this);
        }
    }
}
